### PR TITLE
skywork reward model

### PR DESCRIPTION
**Pull request by Asher-hss**
## Description

add skywork reward model support

## Motivation and Context

add new reward model called skywork

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/models/reward/skywork_model.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        url: Optional[str] = None,
+    ) -> None:
+        device = "cuda:0" if torch.cuda.is_available() else "cpu"
+        super().__init__(model_type, api_key, url)
+        self._client = AutoModelForSequenceClassification.from_pretrained(
+            model_type,
+            torch_dtype=torch.bfloat16,
+            device_map=device,
+            attn_implementation="flash_attention_2",
+            # device_map="auto",
```
Comment (Wendong-Fan) [inline camel/models/reward/skywork_model.py:47]: could you help me understand why not just set `device_map="auto"` here

File: camel/models/reward/skywork_model.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            # offload_folder="offload",
```
Comment (Wendong-Fan) [inline camel/models/reward/skywork_model.py:48]: clean

File: camel/models/reward/skywork_model.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        )
+        with torch.no_grad():
+            score = self._client(inputs).logits[0][0].item()
+            return {"Score": score}
+
+    def get_scores_types(self) -> List[str]:
+        """get the scores types
+        Returns:
+            List[str]: list of scores types
+        """
```
Comment (Wendong-Fan) [inline camel/models/reward/skywork_model.py:75]:
```suggestion
        r"""get the scores types
        
        Returns:
            List[str]: list of scores types
        """
```

File: flash-attention
Comment (Wendong-Fan) [inline flash-attention:1]: clean this file

**Review by Wendong-Fan — COMMENTED** [review]
Thanks @Asher-hss , left some comments below, could you also add unit test?

**Comment by Asher-hss** [comment]
Thanks @Wendong-Fan, These parameters are retained to facilitate our further communication and will be cleared after confirmation.

**Review by Wendong-Fan — APPROVED** [review]
